string action_desc
string action_name
int8 action_flag